### PR TITLE
[FLINK-26219][slf4j] Migrate tests to JUnit5 

### PR DESCRIPTION
**Pull request by zentol**
Based on https://github.com/apache/flink/pull/18817.

### CODE REVIEW

File: flink-metrics/flink-metrics-slf4j/src/test/java/org/apache/flink/metrics/slf4j/Slf4jReporterTest.java [hunk 1]
```diff
@@ -82,17 +77,17 @@ public void testAddCounter() throws Exception {
...
 
     @Test
-    public void testAddGauge() throws Exception {
+    void testAddGauge() throws Exception {
         String gaugeName = "gauge";
 
         Gauge<Long> gauge = () -> null;
         reporter.notifyOfAddedMetric(gauge, gaugeName, metricGroup);
-        assertTrue(reporter.getGauges().containsKey(gauge));
+        assertThat(reporter.getGauges().containsKey(gauge));
```
Comment (RyanSkraba) [inline flink-metrics/flink-metrics-slf4j/src/test/java/org/apache/flink/metrics/slf4j/Slf4jReporterTest.java:90]:
```suggestion
        assertThat(reporter.getGauges()).containsKey(gauge);
```
Misplaced parenthesis causes this test to be ignored.

File: flink-metrics/flink-metrics-slf4j/src/test/java/org/apache/flink/metrics/slf4j/Slf4jReporterTest.java [hunk 2]
```diff
@@ -101,16 +96,17 @@ public void testAddGauge() throws Exception {
...
 
     @Test
-    public void testAddMeter() throws Exception {
+    void testAddMeter() throws Exception {
         String meterName = "meter";
 
         Meter meter = new MeterView(5);
         reporter.notifyOfAddedMetric(meter, meterName, metricGroup);
-        assertTrue(reporter.getMeters().containsKey(meter));
+        assertThat(reporter.getMeters().containsKey(meter));
```
Comment (RyanSkraba) [inline flink-metrics/flink-metrics-slf4j/src/test/java/org/apache/flink/metrics/slf4j/Slf4jReporterTest.java:109]:
```suggestion
        assertThat(reporter.getMeters()).containsKey(meter);
```

File: flink-metrics/flink-metrics-slf4j/src/test/java/org/apache/flink/metrics/slf4j/Slf4jReporterTest.java [hunk 3]
```diff
@@ -119,31 +115,32 @@ public void testAddMeter() throws Exception {
...
 
     @Test
-    public void testAddHistogram() throws Exception {
+    void testAddHistogram() throws Exception {
         String histogramName = "histogram";
 
         Histogram histogram = new TestHistogram();
         reporter.notifyOfAddedMetric(histogram, histogramName, metricGroup);
-        assertTrue(reporter.getHistograms().containsKey(histogram));
+        assertThat(reporter.getHistograms().containsKey(histogram));
```
Comment (RyanSkraba) [inline flink-metrics/flink-metrics-slf4j/src/test/java/org/apache/flink/metrics/slf4j/Slf4jReporterTest.java:128]:
```suggestion
        assertThat(reporter.getHistograms()).containsKey(histogram);
```

**Review by RyanSkraba — COMMENTED** [review]
A couple of comments, but everything else LGTM: no unnecessary `public` for classes and test methods, no `org.junit` imports that aren't `jupiter`, no matchers that aren't `org.assertj` and the `TestLogger extension` is no longer used.

File: flink-metrics/flink-metrics-slf4j/src/test/java/org/apache/flink/metrics/slf4j/Slf4jReporterTest.java (hunk 2 above)
Comment (zentol) [reply flink-metrics/flink-metrics-slf4j/src/test/java/org/apache/flink/metrics/slf4j/Slf4jReporterTest.java]: oh boy this is gonna be fun to double-check in the future :/

**Pull request merged**